scout/server/blueprints/institutes/controllers.py

```diff
@@ -24,6 +30,66 @@
 TRACKS = {None: "Rare Disease", "rare": "Rare Disease", "cancer": "Cancer"}
 
 
+def causatives(institute_obj, request):
+    """Create content to be displayed on institute causatives page
+
+    Args:
+        institute_obj(dict) An institute object
+        request(flask.request) request sent by user's browser
+
+    Returns:
+        data(dict)
+    """
+    # Retrieve variants grouped by case
+    query = request.args.get("query", "")
+    hgnc_id = None
+    if "|" in query:
+        # filter accepts an array of IDs. Provide an array with one ID element
+        try:
+            hgnc_id = [int(query.split(" | ", 1)[0])]
+        except ValueError:
+            flash("Provided gene info could not be parsed!", "warning")
+
+    variants = list(store.check_causatives(institute_obj=institute_obj, limit_genes=hgnc_id))
+    if variants:
+        variants = sorted(
+            variants,
+            key=lambda k: k.get("hgnc_symbols", [None])[0] or k.get("str_repid") or "",
+        )
+
+    all_variants = {}
+    all_cases = {}
+    for variant_obj in variants:
+        update_representative_gene(variant_obj, variant_obj.get("genes", []))
```

Comment:
This is the only new line of code, and it is used to collect the hgvs_identifier and hgvsp_identifier shown on the table